docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+the following information required to create the `Plan` object is obtained:
+1. Name of the plan.
+2. Workout index numbers in the workout list separated by comma.<br><br>
 
-## Product scope
-### Target user profile
+Once, those information are obtained, the name of the plan to be created will be validated via calling the
+`PlanList#checkPlanNameValidity(userPlanNameInput, className)`.
+This is to ensure all plan names are acceptable and unique in the application.
+If the plan name is invalid, an `InvalidPlanException` exception will be thrown.
+<br><br>
+**(Steps 7.3 to 7.4)** The `PlanList` object will then find out the number of workouts
+to be added into the new plan. Subsequently, it will call the `PlanList#checkMinMaxNumberOfWorkouts(numberOfWorkoutsInAPlan, className)` to ensure
+that the number of workouts to be added into the new plan does not exceed 10 workouts, and there should minimally
+be 1 workout in a plan. If the new plan does not meet the minimum and maximum workout number requirement,
+an `InvalidPlanException` will be thrown.
+<br><br>
+**(Steps 7.5 to 7.6)** An ArrayList of Workout object is created to store the workouts to be added into the new plan.
+<br><br>
+**(Steps 7.7 to 7.10)** As the workout index numbers in the user input (e.g. "1, 2, 3") is of type `String`, 
+the loop will split (by comma) and convert each number string into an `Integer`. 
+Subsequently, the `PlanList#checkWorkoutNumberWithinRange(workoutIndexNumberInteger, className)` method
+is called to ensure that each workout index is within the application's workout list range. 
+Once the check is done, the valid `Workout` object is fetched from the application's workout list based 
+on the workout index and then added into the `ArrayList` that was created in the previous step (Steps 7.5 to 7.6).
+The loop will continue until all workouts to be added in the new plan is added into that `ArrayList`.
+<br><br>
+**(Steps 7.11 to 7.12)** Before adding this new plan, the 
+`PlanList#checkPlanWithSameWorkoutSequence(workoutsToAddInAPlanList, className)` is called to ensure that
+the new plan to be created does not contain the same workout order as any existing plans. If it does contain
+the same workout order as an existing plan, an `InvalidPlanException` exception will be thrown.
+<br><br>
+**(Steps 7.13 to 7.18)** With the valid plan name and the `ArrayList` containing the workouts to be added into 
+the new plan, a new `Plan` object is created. This new `Plan` object is then added to the application's plan list.
+<br><br>
+**(Step 8)** The `PlanList#createAndAddPlan(userArgument)` method returns the new `Plan` object to `PlanCommand`.
+<br><br>
+**(Steps 9 to 11)** Upon returning to the `PlanCommand` object, the `UI#printNewPlanCreatedMessage(newPlan)` is called
+to display the plan that has been created to the user via the terminal. The following is an example
+of a success plan creation message (new plan is called "Grow My Muscles"):
```

Comment:
Good and detailed explanation of the sequence diagram, perhaps you can consider adding some examples as well.